lib/swaps/Swaps.ts

```diff
@@ -102,16 +104,20 @@ class Swaps extends EventEmitter {
    */
   public static calculateInboundOutboundAmounts = (quantity: number, price: number, isBuy: boolean, pairId: string) => {
     const [baseCurrency, quoteCurrency] = pairId.split('/');
-    const baseCurrencyAmount = Math.round(quantity * Swaps.UNITS_PER_CURRENCY[baseCurrency]);
+    const baseCurrencyAmount = quantity;
     const quoteCurrencyAmount = price > 0 && price < Number.POSITIVE_INFINITY ?
-      Math.round(quantity * price * Swaps.UNITS_PER_CURRENCY[quoteCurrency]) :
+      Math.round(quantity * price) :
       0; // if price is zero or infinity, this is a market order and we can't know the quote currency amount
+    const baseCurrencyUnits = Math.floor(baseCurrencyAmount * Swaps.UNITS_PER_CURRENCY[baseCurrency]);
+    const quoteCurrencyUnits = Math.floor(quoteCurrencyAmount * Swaps.UNITS_PER_CURRENCY[quoteCurrency]);
 
     const inboundCurrency = isBuy ? baseCurrency : quoteCurrency;
     const inboundAmount = isBuy ? baseCurrencyAmount : quoteCurrencyAmount;
+    const inboundUnits = isBuy ? baseCurrencyUnits : quoteCurrencyUnits;
     const outboundCurrency = isBuy ? quoteCurrency : baseCurrency;
     const outboundAmount = isBuy ? quoteCurrencyAmount : baseCurrencyAmount;
-    return { inboundCurrency, inboundAmount, outboundCurrency, outboundAmount };
+    const outboundUnits = isBuy ? quoteCurrencyAmount : baseCurrencyAmount;
```

Comment:
I believe this is incorrect. Did you mean `quoteCurrencyUnits` and `baseCurrencyUnits`?

Reply:
Yes that's right, good catch.